### DISCUSSION TITLE
No Sound

### DISCUSSION THREAD

**Discussion by DATuttleJr** [music-assistant/support#1913] [Q&A] [answered]
I have HA installed on a Pi4. It is running MASS Beta. I am trying to get the music part of it to work. Here is what I'm experiencing:

I have my laptop open and logged into HAlocal. I have YT Music showing up. I have some local songs as well. I try to play Music. I get no sound at all. 

I also have my Pixel phone open and using Google Home. I can see that my something is casting the requested song but still no sound. If I stop casting from HA I can cast from my phone and I hear all the sounds. I can also go on my laptop and see in HA that the speakers are playing songs from my phone.  Actually now in MASS I can see "External source active: YouTube Music". 

While music is playing from my phone, if I go to a dashboard and use the media player I can pause music, play music again, even skip songs in playlist and it shows song info. If I click on top right corner I can adjust volume and the previous things. If I try to change media it only shows locally stored music.

Go back into MASS in HA and play a new song from YT Music and it shows playing, on my phone, it even gives the name of the new song, but no music. If I go to dashboard media player it shows new song data but you press play and nothing happens. The log says that it is idle. It will not go to playing.

**Comment by OzGav** [comment]
MA has to send the music to a supprted player. It doesn't just play locally. If you want to play to your laptop then you could install the companion app, you could run a squeezelite client or you could run a snapcast server on your network and use snapweb. More info is available in the FAQs https://github.com/orgs/music-assistant/discussions/710

If you have a Google chromecast device then that should be automatically discovered by MA and you should be able to cast to it.

The way you are decribing what you are doing is not clear. It sounds like you are running MA as an addon within HA which is good. You should now access the MA UI from the HA SETTINGS>>ADDONS>>MUSIC ASSISTANT. Once there you add your music providers which sounds like it shold be YouTube Music and the local provider. You should review MA SETTINGS>>PLAYERS and see what players have been discovered, please tell us what they are. You now should be able to select a track (for example) and select PLAY and ensure you select the correct player in the PLAY ON field.

**Comment by DATuttleJr** [comment]
There is one called "Lodge Speakers" this is actually a group made in Google Home. It is two Audiopro C10 MKII speakers which support Google Cast and Air Play. I am mostly in the Googleverse so I didn't plan on using Air Play.

**Reply by OzGav** [reply]
Those individual Audiopro speakers should be detected as well. Assuming you have successfully added the music providers to MA and you can see information in the UI then what happens when you play to the speakers which have been discovered,

**Comment by DATuttleJr** [comment]
I believe I have added my YT Music properly. It shows all my playlists and songs and such in MA. If I choose a song in either YT or Local it shows up on the bottom "player", I choose the Lodge Speakers but when I press play NOTHING happens

**Reply by OzGav** [reply]
Look in the logs for errors. If the indivudal speakers aren't showing up then something is wrong. Do you have a standard network setup, no subnets/VLANs/firewalls

**Comment by DATuttleJr** [comment]
I have a Main Network, IoT Network and a Guest Network. My HA is on my Main Network. My speakers are on my IoT network. My phone is on my main network as well and it can cast to those speakers.

**Reply by OzGav** [reply]
Thats why it isn't working. For you to fix this you will need to move HA to the IOT network. Read here for more info https://github.com/orgs/music-assistant/discussions/709

**Reply by DATuttleJr** [reply]
The speakers show up individually in HA but not in MA.

**Reply by OzGav** [reply]
Not relevant unfortunately

**Reply by OzGav** [reply, marked as answer]
Some people have had success with mDNS forwarders but we do not provide any support for non-standard network setups

**Reply by DATuttleJr** [reply]
![Network setup](https://github.com/music-assistant/support/assets/47952682/57769bbf-c2dc-4321-a912-0af79ee43df8)

**Reply by OzGav** [reply]
People have successes and failures with UniFi gear. Hopefully you can figure it out.
